cmd/kubelet/app/options/options.go

```diff
@@ -328,8 +329,8 @@ func (f *KubeletFlags) AddFlags(mainfs *pflag.FlagSet) {
 	fs.MarkDeprecated("keep-terminated-pod-volumes", "will be removed in a future version")
 	fs.StringVar(&f.ExperimentalMounterPath, "experimental-mounter-path", f.ExperimentalMounterPath, "[Experimental] Path of mounter binary. Leave empty to use the default mount.")
 	fs.MarkDeprecated("experimental-mounter-path", "will be removed in 1.25 or later. in favor of using CSI.")
-	fs.StringVar(&f.CloudProvider, "cloud-provider", f.CloudProvider, "The provider for cloud services. Set to empty string for running with no cloud provider. If set, the cloud provider determines the name of the node (consult cloud provider documentation to determine if and how the hostname is used).")
-	fs.MarkDeprecated("cloud-provider", "will be removed in 1.25 or later, in favor of removing cloud provider code from Kubelet.")
+	fs.StringVar(&f.CloudProvider, "cloud-provider", f.CloudProvider, fmt.Sprintf("The provider for cloud services. Set to empty string for running with no cloud provider. If set, the cloud provider determines the name of the node (consult cloud provider documentation to determine if and how the hostname is used).  If set to 'external', then kubelet will taint nodes with %s.", cloudproviderapi.TaintExternalCloudProvider))
```

Comment:
I included the reference to `TaintExternalCloudProvider` here, as #110066 did, but I'm not sure we should elaborate on the specific behavior toggled with this option -- probably better to have that elsewhere: https://github.com/kubernetes/website/blob/9137c09b0a38bcbe248e749458542931c9b28c93/content/en/docs/tasks/administer-cluster/running-cloud-controller.md?plain=1#L34-L44